platform/src/main/java/org/hillview/storage/CassandraDatabase.java

```diff
@@ -136,24 +165,64 @@ public TokenRange (String rawEntry) {
         }
 
         public String toString() {
-            return "start: " + this.start_token + ", end: " + this.end_token +
-                ", endpoints: " + this.endpoints.toString();
+            StringBuilder sb = new StringBuilder();
+            sb.append("tokenRange: " + this.tokenRange.toString());
+            sb.append(", endpoints: " + this.endpoints.toString());
+            return sb.toString();
+        }
+    }
+
+    private String convertEndpointToIP(String endpoint) {
+        String IPAddress = null;
+        try {
+            InetAddress inet = InetAddress.getByName(endpoint);
+            IPAddress = inet.getHostAddress();
+        } catch (UnknownHostException e) {
+            throw new RuntimeException(e);
+        }
+        return IPAddress;
+    }
+
+    private String discoverLocalEndpoint() throws Exception {
+        Path cassandraPath = Paths.get(this.info.cassandraRootDir);
+        Path cassandraYaml = Paths.get(cassandraPath.toString(), CassandraDatabase.cassandraYamlPath);
+        String nodeName = null;
+        File cassandraConfig = cassandraYaml.toFile();
+        Scanner myReader = new Scanner(cassandraConfig);
+        while (myReader.hasNextLine()) {
```

Comment:
isn't there a library to read yaml so you don't need to do this in this way?

Reply:
There is, but it will load all config parameters. We just need one parameter in the beginning, so I was thinking to get it manually to save space and time. For now, I will update this to use yaml-reader library